### DISCUSSION TITLE
How to deal with mutiboot system not displaying all/correct snapshots.

### DISCUSSION THREAD

**Discussion by iconoclasthero** [Antynea/grub-btrfs#349] [Q&A]
**tl;dr:**  Is there a way to get nvme0n1p9/boot/grub/grub.cfg to also display nvme0n1p10/boot/grub/grub-btrfs.cfg that does not involve any re-partitioning?

---

OS 0: Windows (never used)
OS 1: Ubuntu 22.04 (it _is_ `do-release-upgrade` day today...)
OS 2: Rhino 2024.1

![Screenshot from 2024-08-29 08-00-32](https://github.com/user-attachments/assets/62705729-b9a7-4248-ab38-21de2c316b05)

There's no /boot partition but there is an /boot/efi partition.

The grub that boots is on nvme0n1p9/ubuntu and when it boots it displays the snapshots for ubuntu.

When I run grub-btrfs and grub-mkconfig on Rhino, the Timeshift snapshots are picked up and the nvme0n1p10/boot/grub/grub-btrfs.cfg is generated.

When I boot, nvme0n1p9/boot/grub/grub-btrfs.cfg is displayed which only has the snapshots for Ubuntu 22.04.

If I concat nvme0n1p10/boot/grub/grub-btrfs.cfg to nvme0n1p9/boot/grub/grub-btrfs.cfg I do get both menus.

I can manually replace nvme0n1p9/boot/grub/grub-btrfs.cfg with nvme0n1p10/boot/grub/grub-btrfs.cfg and get my Rhino snapshots.

This is really not a sustainable way to deal with snapshots so I'm wondering if there's a way to get nvme0n1p9/boot/grub/grub.cfg to also display nvme0n1p10/boot/grub/grub-btrfs.cfg that does not involve any re-partitioning, e.g., to add a /boot partition (which I'll do next time...).

FWIW, I really don't know what is at nvme0n1p8/grub2 core.img.  I don't think it's being used by anything at this point.

It looks like if I had to, I could cut out a GB from the windows partition and make a separate /boot, but i really don't want to frack around with changing boot partitions...  I do want to be able to boot into windows should I ever need to—though that ship might have sailed.
